web/server/tests/unit/test_metadata_parser.py

```diff
@@ -0,0 +1,213 @@
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+""" Unit tests for the metadata parser. """
+
+import os
+import unittest
+
+from codechecker_server.metadata import MetadataInfoParser
+
+
+metadata_cc_info = {
+    'check_commands': [
+        'CodeChecker.py analyze -o /path/to/reports /path/to/build.json'
+    ],
+    'timestamps': [1.0],
```

Comment:
Shouldn't the `timestamps` be a dict with `begin` and `end` keys?

Reply:
No. Here we will test the return value of the `__get_metadata_info_v1` and `__get_metadata_info_v2` functions. In case of timestamps it will return a list where each item will be calculated in the following way: 
https://github.com/csordasmarton/codechecker/blob/8b561a529e73b47f28cea600ca8264b8125f7f62/web/server/codechecker_server/metadata.py#L35-L38

The real metada file can be found here: https://github.com/csordasmarton/codechecker/blob/8b561a529e73b47f28cea600ca8264b8125f7f62/web/server/tests/unit/metadata_test_files/v1.json#L40-L42

And the value which can be found here 1.0 = 1571297868 - 1571297867

Reply:
Ok, but in that case this not a timestamp but some duration value or time difference calculated from two timestamps.
For each check command there will be a value:
```
    'check_commands': [
        'CodeChecker.py analyze -o /path/to/reports /path/to/build.json',
        'cppcheck /path/to/main.cpp'
    ],
    'timestamps': [1.0, 1.0],
```
and with the index from the `check_commands` list we can get the corresponding analysis duration from the
`timestamps` list.
If that is right the something like `analysis_duration` would be a bit better name. Only the duration is stored not the start and end dates in the database right?